### PR TITLE
add `azurerm_nginx_certificate` data source

### PR DESCRIPTION
**Pull request by puneetsarna**
```shell
terraform-provider-azurerm $ make acctests SERVICE="nginx" TESTARGS="-run=TestAccNginxCertificateDataSource_basic"
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/nginx -run=TestAccNginxCertificateDataSource_basic -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNginxCertificateDataSource_basic
=== PAUSE TestAccNginxCertificateDataSource_basic
=== CONT  TestAccNginxCertificateDataSource_basic
--- PASS: TestAccNginxCertificateDataSource_basic (626.47s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx	629.577s
```

### CODE REVIEW

File: internal/services/nginx/nginx_certificate_data_source.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Nginx.NginxCertificate
+			var model CertificateDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+			deploymentId, err := nginxdeployment.ParseNginxDeploymentID(model.NginxDeploymentId)
+			if err != nil {
+				return fmt.Errorf("error parsing NGINX deployment ID %s: %+v", deploymentId, err)
```
Comment (stephybun) [inline internal/services/nginx/nginx_certificate_data_source.go:86]: The error returned by the parsing function already contains the information we need. We don't need to wrap it further.
```suggestion
				return err
```

File: internal/services/nginx/nginx_certificate_data_source.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+				deploymentId.ResourceGroupName,
+				deploymentId.NginxDeploymentName,
+				model.Name,
+			)
+			result, err := client.CertificatesGet(ctx, id)
+			if err != nil {
+				if response.WasNotFound(result.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+				return fmt.Errorf("reading %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/nginx/nginx_certificate_data_source.go:99]:
```suggestion
				return fmt.Errorf("retrieving %s: %+v", id, err)
```

File: internal/services/nginx/nginx_certificate_data_source.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+			output := CertificateDataSourceModel{
+				Name:              id.CertificateName,
+				NginxDeploymentId: deploymentId.ID(),
+			}
+
+			if model := result.Model; model != nil {
+				prop := result.Model.Properties
+				output.KeyVirtualPath = pointer.ToString(prop.KeyVirtualPath)
+				output.KeyVaultSecretId = pointer.ToString(prop.KeyVaultSecretId)
+				output.CertificateVirtualPath = pointer.ToString(prop.CertificateVirtualPath)
```
Comment (stephybun) [inline internal/services/nginx/nginx_certificate_data_source.go:111]: We should be using the generic `pointer.From` here and not `.ToString`
```suggestion
				output.KeyVirtualPath = pointer.From(prop.KeyVirtualPath)
				output.KeyVaultSecretId = pointer.From(prop.KeyVaultSecretId)
				output.CertificateVirtualPath = pointer.From(prop.CertificateVirtualPath)
```

File: website/docs/d/nginx_certificate.html.markdown [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+* `nginx_deployment_id` - (Required) The ID of the Nginx Deployment that this certificate is associated with.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Nginx Certificate.
+
+* `certificate_virtual_path` - The path to the cert file of this certificate.
```
Comment (stephybun) [inline website/docs/d/nginx_certificate.html.markdown:40]:
```suggestion
* `certificate_virtual_path` - The path to the certificate file of this certificate.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this @puneetsarna.

I left a few minor comments that should be fixed up for consistency across the provider. Once that's done we can run the tests and this should be good to go!

File: internal/services/nginx/nginx_certificate_data_source.go (hunk 1 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_certificate_data_source.go]: 👍 made the change.

File: internal/services/nginx/nginx_certificate_data_source.go (hunk 2 above)
Comment (puneetsarna) [reply internal/services/nginx/nginx_certificate_data_source.go]: done 👍

File: website/docs/d/nginx_certificate.html.markdown (hunk 4 above)
Comment (puneetsarna) [reply website/docs/d/nginx_certificate.html.markdown]: 👍 made the change.

**Comment by puneetsarna** [comment]
@stephybun Thank you for the review. I have made the suggested changes and have rerun the acceptance test as well.

**Review by stephybun — APPROVED** [review]
Thanks @puneetsarna LGTM 🍕

**Pull request merged**